### ISSUE TITLE
FID for ffhq 1024 

### ISSUE THREAD

**Issue by betterze** [ericl122333/PatchDiffusion-Pytorch#1]
Dear PatchDiffusion group,

Thank you for sharing this work, I really enojy it. 

Could you tell me the FID for ffhq 1024 of the patch diffusion model? 

Thank you for your help.

Best Wishes,

Zongze

**Comment by ericl122333** [comment]
Hello Zongze,

Thank you for reaching out. Unfortunately, we to date have not actually computed a FID metric for FFHQ. 

Aside from compute constraints, our primary reason for this is because of the downsampling used during FID evaluation. Specifically, Inception resizes the images to 299x299, which gets rid of the highest-resolution details for 1024x1024 images. This is especially important for evaluating patched diffusion models, since they tend to struggle most with these details.

That being said, there are likely other quantitative metrics for evaluating our approach (e.g. FID on random image crops, frechet distance with a different network besides inception, etc.). 

Best,
Eric

**Comment by betterze** [comment]
Dear Eric,

Thank you for your reply.

If I understand correctly, the ffhq 1024 model is trained in 14 V-100 days, which is very short. Have you ever try to train it for longer time untill it converge? Your model architecture is based on guided diffusion model, have you try to train guided diffusion model in ffhq 1024 and compare its performance with your model？

Thank you for your help.

Zongze

**Comment by ericl122333** [comment]
Hello Zongze, 

Thank you for your reply and your interest in our work. While our FFHQ model was indeed trained for only about 14 V-100 days (~7 TPUv2-8 days), this actually amounts to a total of 25 million training images (800k iters * 32 img/iter). For reference, this is the same amount of images that the StyleGan2 generator saw during training (Table 1 of [1]). The fast training time is partly due to the low theoretical FLOP usage of our patched architecture, and partly due to the better hardware utilization that comes with our larger batch size.

We have not trained a regular guided diffusion model on FFHQ 1024 due to memory constraints; our available VRAM amounts to only 64 GB per session. It is likely that for a given training time, our model will have better global structure than a normal ADM model, but comparably worse high-level detail. To prioritize low-level features, one can always switch from our "square root SNR" loss weighting to the normal SNR loss weighting used in ADM (if doing this, we recommend using the v-parameterization from [2])

Regards,
Eric

References: 
[1] https://arxiv.org/pdf/1912.04958.pdf
[2] https://arxiv.org/pdf/2202.00512.pdf

**Comment by betterze** [comment]
Thank you for your detailed reply. I really appreciate it. Thx a lot.
